### ISSUE TITLE
Investigate further third-party OSS libraries

### ISSUE THREAD

**Issue by remmeier** [stagemonitor/stagemonitor#172]
The use of Dropwizard Metrics and Zipkin could give stagemonitor quite a boost.

**Comment by felixbarny** [comment]
Stagemonitor is already using dropwizard metrics

**Comment by felixbarny** [comment]
Distributed tracing is definitely something I'm considering but as this is already discussed in #118 I'm closing this. 

Thank you very much for your input! Feel free to make further suggestions or to explain how you think the integration should be done.

**Comment by remmeier** [comment]
ok, great. At least to be it has not been apparent from neither the stagemonitor nor the dropwizard page. I studied the code a bit. Maybe some suggestions:

- have you considered the dropwizard healthcheck api for alerting?
- http://metrics.dropwizard.io/3.1.0/manual/third-party/ has a list of third-party contributions, maybe a couple of the stagemonitor metric libraries could be listed there, like stagemonitor-os and stagemonitor-jdbc. Ideally those components would be fairly lightweight and can work without the complete stagemonitor stack (such libraries could still use provided dependencies to get the full feature set when use with stagemonitor).
- I also noticed that stagemonitor has custom reporters to elasticsearch and influxdb. There are also some plugins mentioned here: http://metrics.dropwizard.io/3.1.0/manual/third-party/. I can't judge which ones do a better job for which use cases. But if the stagemonitor ones have some advantages, they could be made available too.
- maybe the metrics web page with the charts could be made available as further dropwizard add-on that runs just with the metrics registry / servlet. Just like the metrics servlets of dropwizard that are fairly lightweight.

The last three questions go a bit into the same direction. We have been using Dropwizard Metrics for a while. It is quite popular in the community, small, pluggable and has many third-party contributions. Which in the end also gives fewer risks and allows to start using and contributing to it incrementally without having to choose the full stack. Maybe that is a path that stagemonitor could do as well by allowing parts of it to be integrated into dropwizard metrics and advertise it accordingly. But it definitely should continue to advertise the complete out-of-the-box solution for java monitoring as well.
